Haptics/app/src/main/java/com/example/android/haptics/samples/ui/views/HomeViewModel.kt

```diff
@@ -0,0 +1,257 @@
+                                    VibrationEffect.Composition.PRIMITIVE_CLICK
+                                ),
+                                HapticFeedbackButton(
+                                    application.getString(R.string.home_screen_slow_rise),
+                                    isPrimitiveSupported(vibrator, VibrationEffect.Composition.PRIMITIVE_SLOW_RISE),
+                                    VibrationEffect.Composition.PRIMITIVE_SLOW_RISE
+                                ),
+                                HapticFeedbackButton(
+                                    application.getString(R.string.home_screen_quick_rise),
+                                    isPrimitiveSupported(vibrator, VibrationEffect.Composition.PRIMITIVE_QUICK_RISE),
+                                    VibrationEffect.Composition.PRIMITIVE_QUICK_RISE
+                                ),
+                                HapticFeedbackButton(
+                                    application.getString(R.string.home_screen_quick_fall),
+                                    isPrimitiveSupported(vibrator, VibrationEffect.Composition.PRIMITIVE_QUICK_FALL),
+                                    VibrationEffect.Composition.PRIMITIVE_QUICK_FALL
+                                ),
+                                HapticFeedbackButton(
+                                    application.getString(R.string.home_screen_spin),
+                                    isPrimitiveSupported(vibrator, VibrationEffect.Composition.PRIMITIVE_SPIN),
+                                    VibrationEffect.Composition.PRIMITIVE_SPIN
+                                ),
+                                HapticFeedbackButton(
+                                    application.getString(R.string.home_screen_thud),
+                                    isPrimitiveSupported(vibrator, VibrationEffect.Composition.PRIMITIVE_THUD),
+                                    VibrationEffect.Composition.PRIMITIVE_THUD
+                                ),
+
+                            )
+                        ),
+                    )
+                )
+                return HomeViewModel(
+                    vibrator = vibrator, viewModelState = viewModelState,
+                    scaffoldState = scaffoldState, scrollState = scrollState
+                ) as T
+            }
+        }
+    }
+}
+
+/**
+ * Query whether the vibrator natively supports the a given effects.
+ * Note that this logic means that either the effect is definitely supported by the hardware,
+ * or it's status is unknown. Even with a value of VIBRATION_EFFECT_SUPPORT_NO, the device would
+ * either fall back to an unoptimized version or no vibration at all.
+ *
+ * See https://developer.android.com/reference/android/os/Vibrator#areEffectsSupported(int...)>
+ * for more information.
+ */
+private fun isEffectSupported(vibrator: Vibrator, effectId: Int): Boolean {
+    if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.R) return false
```

Comment:
My suggestion is to only check if API is Q+, since the code was introduced then. See my comment below.

Reply:
I think I updated to what I understand you to mean.